src/subsystems/haskell/translators/cabal-freeze/default.nix

```diff
@@ -276,5 +277,12 @@ in {
       examples = ["9.0.2" "9.4.1"];
       type = "argument";
     };
+
+    defaultPackageName = {
+      description = "Name of default package";
+      default = "main";
+      examples = ["main" "default"];
+      type = "argument";
+    };
```

Comment:
Is there a way of setting the package name without adding a new entry to `extraArgs`?

Reply:
Maybe regex the .cabal files for the name and version?
Alternatively cabal2json could be used but this introduces a layer of IFD.